lib/workload/stateless/stacks/fmannotator/api.go

```diff
@@ -0,0 +1,94 @@
+// Package fmannotator This file contains definitions for calling the FileManager API.
+package fmannotator
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/umccr/orcabus/lib/workload/stateless/stacks/fmannotator/schema/orcabus_workflowmanager/workflowrunstatechange"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+)
+
+// A JsonPatch used to update FM records.
+type JsonPatch struct {
+	Op    string `json:"op"`
+	Path  string `json:"path"`
+	Value string `json:"value,omitempty"`
+}
+
+// The PatchList for updating records.
+type PatchList []JsonPatch
+
+// MarshallPortalRunId MarshallPatch Convert an event into a JSON patch using the portalRunId.
+func MarshallPortalRunId(event *workflowrunstatechange.Event) ([]byte, error) {
+	return json.Marshal(PatchList{JsonPatch{
+		"add",
+		"/portalRunId",
+		event.Detail.PortalRunId,
+	}})
+}
+
+// The ApiClient which is used to send requests to filemanager.
+type ApiClient struct {
+	Client  *http.Client
+	Request *http.Request
+}
+
+// NewApiClient Create a new ApiClient.
+func NewApiClient(config *Config, body io.Reader) (*ApiClient, error) {
+	req, err := http.NewRequest("GET", config.FileManagerEndpoint, body)
+	if err != nil {
```

Comment:
No way to have those as a simple macro or similar in Go? Tons of 3 line repetition :-S

Reply:
Other than putting the `if` condition on the same line:

```golang
req, err := http.NewRequest("GET", config.FileManagerEndpoint, body); if err != nil {
    return err
}
```

The vibe online seems to be to just embrace it :)

Reply:
"Luckily, Go does not support macros." - https://stackoverflow.com/questions/49527584/how-to-implement-macros-in-go